plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -321,10 +322,12 @@ func getFitPredicateFunctions(names sets.String, args PluginFactoryArgs) (map[st
 		predicates[name] = factory(args)
 	}
 
-	// Always include required fit predicates.
-	for name, factory := range mandatoryFitPredicateMap {
-		if _, found := predicates[name]; !found {
-			predicates[name] = factory(args)
+	// Always include mandatory fit predicates.
+	for name, mandatory := range mandatoryFitPredicates {
+		if mandatory {
```

Comment:
The current logic is correct, but I would remove checking the value of `mandatory`. `mandatoryFitPredicate` is basically a set. Anything in that set can be considered a mandatory predicate, regardless of the value of `bool`.